lib/core/test/unit/Cardano/Wallet/Primitive/TypesSpec.hs

```diff
@@ -447,6 +453,51 @@ spec = do
                             . slotStartTime sps
                     Just slot === f slot
 
+    describe "Converting Range SlotId to Range UTCTime and back" $ do
+        let f sps r = fmap (slotStartTime sps) <$> slotRange sps r
+        let sharedDescription = " (where f = fmap slotStartTime . slotRange)"
+        it ("f timeRange ⊆ timeRange" ++ sharedDescription) $
+            withMaxSuccess 1000 $ property $
+                \sps timeRange -> let r = getUniformTime <$> timeRange in
+                    cover 10 (isJust $ f sps r) "slot range exists" $ do
+                    case f sps r of
+                        Just r' -> r' `isSubrangeOf` r
+                        Nothing -> True
+        it ("f^n === f" ++ sharedDescription) $
+            withMaxSuccess 1000 $ property $
+                \sps timeRange (NonNegative n) -> do
+                    let r = getUniformTime <$> timeRange
+                    let r' = f sps r
+                    iterate (>>= f sps) r' !! n === r'
```

Comment:
Does the existence of `Data.Function.Utils.applyN` make the above property easier to define?

Reply:
Reads better at least 👍

Reply:
Great!